components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -232,27 +245,110 @@ impl IndexScanExecutorImpl {
                 return Err(other_err!("Unexpected missing column {}", col_id));
             }
         }
+        Ok(())
+    }
+
+    fn extract_pk_columns_from_datum_payload(
+        &mut self,
+        payload: &mut &[u8],
+        columns: &mut LazyBatchColumnVec,
+        start: usize,
+        end: usize,
+        ignore_value: bool,
+    ) -> Result<()> {
+        for i in start..end {
+            if payload.is_empty() {
+                break;
+            }
+            let (value, remain) = datum::split_datum(payload, false)?;
+            if !ignore_value {
+                columns[i].mut_raw().push(value);
+            }
+            *payload = remain;
+        }
+        Ok(())
+    }
+
+    fn process_unique_common_handle_value(
+        &mut self,
+        key: &[u8],
+        mut value: &[u8],
+        columns: &mut LazyBatchColumnVec,
+    ) -> Result<()> {
+        let tail_len = value[0] as usize;
```

Comment:
The case `tail_len == 1` only exists in TiDB memory.
All values in TiKV has tail_len == 0.